### PR TITLE
Improve navigation section pages

### PR DESCRIPTION
**Pull request by skyace65**
Supersedes #6044 and #6053. Per discussions on Godot's rocket.chat the content of those two PRs has been combined to replace the existing 3D introduction page and create a 2D introduction page. @mhilbrunner I know you were waiting on this. I've made some minor tweaks to the content from those PR's and copied over the RVO section from the existing introduction page. @smix8 please look this over as well and see if there's anything that should be changed or fixed, and thank you for the two PR's and other navigation ones, they're a big help.

### CODE REVIEW

**Comment by YuriSizov** [comment]
I'm assuming that smix8 should be added as a co-author for this one then?

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
+.. _doc_navigation_overview:
```
Comment (smix8) [inline tutorials/navigation/navigation_introduction_3d.rst:1]: `navigation_introduction_3d` or `navigation_overview`, what shall it be? Same for 2D.

Reminder for the need to check the linking of other pages in the open pr's if this gets changed around, can't remember if this page was crosslinked.

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+However agents can only take into account a set number of other nearby
+agents, this is the :ref:`max neighbors<class_NavigationAgent3D_property_max_neighbors>`
+property of an agent and can be adjusted. This is **not** a limit for
+how many agents can use a navigation region at the same time.
+
+Setup for 3D scene
+------------------
+
+The following steps show how to setup a minimum viable RVO navigation in 3D that uses the NavigationServer3D and
```
Comment (smix8) [inline tutorials/navigation/navigation_introduction_3d.rst:93]: This is not correct as the following steps and examples do not include anything about RVO avoidance. The original was "how to setup a minimum viable navigation in 3D".

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+- :ref:`NavigationMesh<class_NavigationMesh>` Resource
+    A resource that holds 3D navigationmesh data and provides 3D geometry baking options to define navigation
+    areas inside the Editor as well as at runtime.
+
+    - The NavigationRegion3D Node uses this resource to define its navigation area.
+    - The NavigationServer3D uses this resource to update navmesh of individual regions.
+    - The GridMap Editor uses this resource when specific navigationmeshes are defined for each gridcell.
+
+RVO processing
```
Comment (smix8) [inline tutorials/navigation/navigation_introduction_3d.rst:72]: I am not sure about including so much about the RVO avoidance in was is more or less a navigation beginners page as it is a very overwhelming / confusing topic for new users and there is no steps or examples for it included here. Maybe only mention it as an available feature for the overview with one sentence / little paragraph and a link to the main avoidance page and merge the detail there? Same for 2D.

**Comment by skyace65** [comment]
I've fixed the issues and removed mention of RVO except for the obstacle nodes, this page should be updated in whatever PR adds RVO information so we can link directly to it. Also didn't know co-authoring was a thing, added smix8 as a co-author.

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 4]
```diff
@@ -0,0 +1,147 @@
+.. _doc_navigation_overview_2d:
+
+Introduction
+===================
+
+Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_2d.rst:6]:
```suggestion
Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation
```

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+and pathfinding for 2D and 3D games. The following section provides a quick overview over all available
+navigation related objects in Godot for 2D scenes and their primary use.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_2d.rst:8]: `object`s is used multiple times and strikes me as a bit odd compared to other Godot docs. Shouldn't we call them nodes and classes? Or are these different?

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+Godot provides the following objects and classes for 2D navigation:
+
+- :ref:`Astar2D<class_Astar2D>`
+    ``Astar2D`` objects provide an option to find the shortest path in a graph of weighted **points**.
+
+    The AStar2D class is best suited for cellbased 2D gameplay that does not require actors to reach any possible position within an area but only predefined, distinct positions.
+
+- :ref:`NavigationServer2D<class_NavigationServer2D>`
+    ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a navigationmesh defined area.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_2d.rst:21]:
```suggestion
    ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a area defined by a navigation mesh.
```
`navigationmesh` is used multiple times in the doc, it should IMO either be `navigation mesh`, `navmesh` or link to a class and then be capitalized as `NavigationMesh` if it is one.

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
+.. _doc_navigation_overview_3d:
+
+Introduction
+===================
+
+Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_3d.rst:6]:
```suggestion
Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation
```

File: tutorials/navigation/navigation_introduction_3d.rst [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    The region can be enabled / disabled.
+    The use in pathfinding can be further restricted through the navigationlayers bitmask.
+    Regions can join their navigationmeshes by proximity for a combined navigationmesh.
+
+-  :ref:`NavigationAgent3D<class_NavigationAgent3D>` Node
+    An optional helper Node to facilitate common NavigationServer3D API calls for pathfinding and avoidance for
+    a Node3D inheriting parent Node.
+
+-  :ref:`NavigationObstacle3D<class_NavigationObstacle3D>` Node
+    A Node that acts as an agent with avoidance radius, to work it just needs to be added under a Node3D
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_3d.rst:57]:
```suggestion
    A Node that acts as an agent with avoidance radius, to work it needs to be added under a Node3D
```

File: tutorials/navigation/navigation_introduction_2d.rst [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    The region can be enabled / disabled.
+    The use in pathfinding can be further restricted through the navigationlayers bitmask.
+    Regions can join their navigationmeshes by proximity for a combined navigationmesh.
+
+-  :ref:`NavigationAgent2D<class_NavigationAgent2D>` Node
+    An optional helper Node to facilitate common NavigationServer2D API calls for pathfinding and avoidance
+    for a Node2D inheriting parent Node.
+
+-  :ref:`NavigationObstacle2D<class_NavigationObstacle2D>` Node
+    A Node that acts as an agent with avoidance radius, to work it just needs to be added under a Node2D
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_introduction_2d.rst:54]:
```suggestion
    A Node that acts as an agent with avoidance radius, to work it needs to be added under a Node2D
```

**Review by mhilbrunner — COMMENTED** [review]
Thanks to both of you for collaborating on this. Some comments below:

**Comment by skyace65** [comment]
Fixed.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Good work, thanks for the collaboration you two, lets get this in and iterate if needed; LGTM.

**Pull request merged**